### PR TITLE
feat: add possibility to set required strength for battle action

### PR DESCRIPTION
**Pull request by kkafar**
# Description

You can now configure strength points required to win the battle.

I placed this option in action configuration (`ActionConfigBundle`) not in object (`GameObjectConfig`). However I can see why thats wrong ;D 

# Checklist

* [ ] Included code to test these changes
* [ ] Updated Jira

### CODE REVIEW

File: configurations/demo-config-2/locations/location-2/objects/bandit-1.json [hunk 1]
```diff
@@ -5,18 +5,19 @@
...
     "type": "battle",
-    "reward": 10
+    "reward": 10,
+    "strength": 19
   },
   "onRightClick": {
     "tag": "info",
     "type": "show-description",
-    "description": "Potrzebujesz przynajmniej 1 siły aby mnie pokonać.",
+    "description": "Potrzebujesz przynajmniej 20 siły aby mnie pokonać.",
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-2/objects/bandit-1.json:14]:
```suggestion
    "description": "Potrzebujesz przynajmniej 20 punktów siły, aby mnie pokonać.",
```

File: configurations/demo-config-2/locations/location-2/objects/bandit-1.json [hunk 2]
```diff
@@ -5,18 +5,19 @@
...
     "assetPath": "bandit.png"
   },
   "onApproach": {
     "tag": "info",
     "type": "show-description",
-    "description": "Potrzebujesz przynajmniej 1 siły aby mnie pokonać.",
+    "description": "Potrzebujesz przynajmniej 20 siły aby mnie pokonać.",
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-2/objects/bandit-1.json:20]:
```suggestion
    "description": "Potrzebujesz przynajmniej 20 punktów siły, aby mnie pokonać.",
```

File: configurations/demo-config-2/locations/location-2/objects/bandit-2.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "onLeftClick": {
+    "tag": "battle",
+    "type": "battle",
+    "reward": 10,
+    "strength": 9
+  },
+  "onRightClick": {
+    "tag": "info",
+    "type": "show-description",
+    "description": "Potrzebujesz przynajmniej 10 siły aby mnie pokonać.",
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-2/objects/bandit-2.json:14]:
```suggestion
    "description": "Potrzebujesz przynajmniej 10 punktów siły, aby mnie pokonać.",
```

File: src/main/java/io/rpg/model/actions/ActionEngine.java [hunk 4]
```diff
@@ -131,7 +131,9 @@ public void onAction(BattleAction action) {
   }
 
   public void onAction(LevelUpAction action) {
-    controller().getPopupController().openTextPopup((action.getPoints() != null ? (action.getPoints() + " points earned already!\n") : "" ) + "Achieved level " + action.newLevel + "!",
+    controller().getPopupController().openTextPopup(
+        (action.getPoints() != null ? (action.getPoints() + " points earned already!\n") : "")
+            + "Achieved level " + action.newLevel + "!",
```
Comment (mhawryluk) [inline src/main/java/io/rpg/model/actions/ActionEngine.java:136]: I think it'll look better if we change all of the popup texts to Polish to match the demo configuration.

File: src/main/java/io/rpg/model/actions/BattleReflexAction.java [hunk 5]
```diff
@@ -18,4 +18,5 @@ public void acceptActionEngine(ActionEngine engine) {
   public int getReward() {
     return reward;
   }
+
```
Comment (mhawryluk) [inline src/main/java/io/rpg/model/actions/BattleReflexAction.java:21]:
```suggestion
```

**Review by mhawryluk — APPROVED** [review]
Great!

File: src/main/java/io/rpg/model/actions/ActionEngine.java (hunk 4 above)
Comment (kkafar) [reply src/main/java/io/rpg/model/actions/ActionEngine.java:136]: IMO in separate PR

**Pull request merged**